src/domain/auth/OidcCallback.tsx

```diff
@@ -8,30 +8,52 @@ import * as Sentry from '@sentry/browser';
 
 import userManager from './userManager';
 import PageWrapper from '../app/layout/PageWrapper';
+import ErrorMessage from '../../common/components/error/Error';
 
 function OidcCallback(props: RouteChildrenProps) {
   const { t } = useTranslation();
 
+  const [callbackMessage, setCallbackMessage] = useState({
+    message: <p>{t('authentication.redirect.text')}</p>,
```

Comment:
Here and later you are putting components into state. Wayback when React used to advise against this 🤔Not sure if times have changed since then:

http://web.archive.org/web/20150419023006/http://facebook.github.io/react/docs/interactivity-and-dynamic-uis.html#what-shouldnt-go-in-state

The refactor here would be to render the message you want on the return line, and save such values into state that you are able to determine what error should be rendered.

Reply:
Thank you for this feedback! Good point! Tried to find the PR where this advice was removed from the react docs, even! 

I think this should be fine now four years after the recommendation was removed from the docs, especially given how the hooks work.